v1/plugins/logs/encoder.go

```diff
@@ -141,18 +145,30 @@ func (enc *chunkEncoder) reset() ([][]byte, error) {
 	// decisions in the last chunk.
 	// 3) Equilibrium: If the chunk size is between 90% and 100% of the user-configured limit, maintain soft limit value.
 
-	if enc.buf.Len() < int(float64(enc.limit)*encHardLimitThreshold) {
+	if enc.bytesWritten < int(float64(enc.limit)*encHardLimitThreshold) {
 		if enc.metrics != nil {
 			enc.metrics.Counter(encSoftLimitScaleUpCounterName).Incr()
 		}
 
 		mul := int64(math.Pow(float64(softLimitBaseFactor), float64(enc.softLimitScaleUpExponent+1)))
+		// this can cause enc.softLimit to overflow into a negative value
```

Comment:
What are the circumstances for a scenario where we reach an overflow? Since the thing we're exponentially increasing is upload bytes, for us to overflow, wouldn't the previous successful reset need have had a soft-limit already terabytes in size? 

This is intuition talking, and not me doing actual calculus, though, so I may be way off in my estimates. It's very likely I'm missing something here, since you've encountered this in your work and had to fix it.

Reply:
As we discussed, I updated the PR to now enforce a maximum configurable limit of `4294967296` instead removing the need to check if the soft limit will ever overflow.